frontend/src/lib/components/LemonTable/types.ts

```diff
@@ -67,4 +67,6 @@ export interface ExpandableConfig<T extends Record<string, any>> {
      * A negative value (like -1) means that the row isn't expandable and that also the expand button cell is skipped.
      */
     rowExpandable?: (record: T) => boolean | number
+    /** Called when row is expanded */
+    onRowExpanded?: (record: T) => null
```

Comment:
A nit, but I suggest instead naming `onRowExpand`, as that'd be more inline with the convention arounds events (e.g. `onClick`, `onLoad`, `onChange`).